src/scss/_state.scss

```diff
@@ -0,0 +1,75 @@
+/// apply message status
+/// @param {List} $states [('success', 'neutral', 'error', 'warning', 'warning-light', 'inform', 'inform-inverse')]
+/// @access private
+@mixin _oMessageApply($states) {
+	@each $state in $states {
+		@if index($o-message-states, $state) and _oMessageSupports($state) {
+			@include _oMessageState($state);
+		}
+	}
+}
+
+/// message status styling
+/// @param {String} $state - the name of the status for the message ('success', 'neutral', 'error', 'warning', 'warning-light', 'inform', 'inform-inverse')
+/// @access private
+@mixin _oMessageState ($state) {
+	.o-message--#{$state} {
+		color: oColorsGetPaletteColor(_oMessageGet('foreground-color', $from: $state));
+		background-color: oColorsGetPaletteColor(_oMessageGet('background-color', $from: $state));
+
+		.o-message__content-main a,
+		.o-message__content-additional a {
+			@include oTypographyLinkCustom(
+				$baseColor: _oMessageGet('foreground-color', $from: $state),
+				$hoverColor: _oMessageGet('foreground-color', $from: $state),
+				$backgroundColor: _oMessageGet('background-color', $from: $state)
+			);
+			border-width: 1px;
+		}
+
+		.o-message__actions__primary {
+			@include oButtons();
+			@include oTypographySans($scale: -1);
+			@include oButtonsTheme($theme: (
+				background: _oMessageGet('background-color', $from: $state),
+				accent: _oMessageGet('foreground-color', $from: $state)
+			));
+			margin-right: oTypographySpacingSize(6);
+		}
+
+		.o-message__actions__secondary {
+			@include oTypographyLinkCustom(
+				$baseColor: _oMessageGet('foreground-color', $from: $state),
+				$hoverColor: _oMessageGet('foreground-color', $from: $state),
+				$backgroundColor: _oMessageGet('background-color', $from: $state)
+			);
+			white-space: nowrap;
+
+			&:hover {
+				border-bottom-color: oColorsGetPaletteColor(_oMessageGet('foreground-color', $from: $state)); // is different to custom link hover by design.
+			}
+
+			&:focus {
+				outline: 3px solid oColorsGetPaletteColor(_oMessageGet('foreground-color', $from: $state));
```

Comment:
Do you know why do we override the focus styles? Odd we do it for the link in o-message but nowhere else.

Reply:
Oh, related, we usually include o-normalise in the demo dependencies.